src/components/MoneyRequestHeaderStatusBar.tsx

```diff
@@ -12,15 +13,21 @@ type MoneyRequestHeaderStatusBarProps = {
 
     /** Whether we show the border bottom */
     shouldShowBorderBottom: boolean;
+
+    /** Badge background color Style */
+    badgeColorStyle?: StyleProp<ViewStyle>;
 };
 
-function MoneyRequestHeaderStatusBar({title, description, shouldShowBorderBottom}: MoneyRequestHeaderStatusBarProps) {
+function MoneyRequestHeaderStatusBar({title, description, shouldShowBorderBottom, badgeColorStyle}: MoneyRequestHeaderStatusBarProps) {
     const styles = useThemeStyles();
     const borderBottomStyle = shouldShowBorderBottom ? styles.borderBottom : {};
+    const badgeBackgroundColorStyle = badgeColorStyle ?? styles.moneyRequestHeaderStatusBarBadgeBackground;
+    const badgeTextColorStyle = badgeColorStyle ? styles.textMicroBoldDangerColor : styles.textMicroBoldColor;
```

Comment:
Why `badgeColorStyle` is used like a boolean here. 
Why if `badgeColorStyle`color exists text style should be styles.`textMicroBoldDangerColor`?

As `badgeColorStyle` could be anything.

Can you please refactor this properly?